app/grandchallenge/reader_studies/forms.py

```diff
@@ -310,6 +322,54 @@ def interface_choices(self):
     def initial_interface(self):
         return self.interface_choices().first()
 
+    def widget_choices(self):
+        answer_type = self["answer_type"].value()
+        choices = [("", "Default")]
+        if answer_type:
+            choices.extend(
+                QuestionWidget.get_widget_choices_for_answer_type(
+                    answer_type=answer_type
+                )
+            )
+        return choices
+
+    def initial_widget(self):
+        try:
+            return self.instance.widget.kind
+        except ObjectDoesNotExist:
+            return ""
+
+    def clean_widget(self):
+        value = self.cleaned_data.get("widget")
+        if value == "":
+            try:
+                self.instance.widget.delete()
+            except ObjectDoesNotExist:
+                pass
+            return None
+        else:
+            widget_instance = QuestionWidget(kind=value)
+
+            if value == QuestionWidgetKindChoices.ACCEPT_REJECT:
+                if not self.cleaned_data.get("interface"):
+                    self.add_error(
+                        error=ValidationError(
+                            f"To use the {widget_instance.get_kind_display()} widget you must define a default answer."
+                        ),
+                        field=None,
+                    )
+                if self.cleaned_data.get("required"):
+                    self.add_error(
+                        error=ValidationError(
+                            f'To use the {widget_instance.get_kind_display()} widget you must uncheck "required".'
+                        ),
+                        field=None,
+                    )
+            if not self.errors:
+                widget_instance.question = self.instance
+                widget_instance.save()
+                return widget_instance
```

Comment:
At the moment, we're creating a new widget every time someone updates the question (and keeps the Accept / Reject widget chosen). I think it might be worth adding a check here if there already is a widget of the same type for that question and if so return that widget rather than saving the new one. For now this check is simple, in the future, if we have options that would get a little more involved.

Reply:
I think we need to change this somehow, we shouldn't create or delete objects during clean as the responsibility of this method is only to check the set of values. We definitely should not create a new object on every question update. Doing this in the save methods would be more appropriate. You may be able to use the save method directly, or you may have to work with the pre and post save signals.